File: Python Labs/com/pattern/memento/ShoppingCartCaretaker.py
class ShoppingCartCaretaker:
    def __init__(self):
        self.memento_stack = []

    def save_memento(self, memento):
        self.memento_stack.append(memento)

    def undo(self):
        if self.memento_stack:
            return self.memento_stack.pop()
        else:
            return None

File: Java Labs/Patterns On Java/src/com/pattern/memento/ShoppingCartCaretaker.java
package com.pattern.memento;

import java.util.Stack;

public class ShoppingCartCaretaker {
    private Stack<ShoppingCartMemento> mementoStack = new Stack<>();

    public void saveMemento(ShoppingCartMemento memento) {
        mementoStack.push(memento);
    }

    public ShoppingCartMemento undo() {
        if (!mementoStack.isEmpty()) {
            return mementoStack.pop();
        } else {
            return null;
        }
    }
}